core/prepare_test.go

```diff
@@ -46,57 +47,64 @@ func testMakePrepareHandlerPrimary(t *testing.T) {
 			ClientId: rand.Uint32(),
 		},
 	}
-	prepareUIBytes := make([]byte, 1)
-	rand.Read(prepareUIBytes)
+	ui := &usig.UI{Counter: rand.Uint64()}
 	makePrepareMsg := func(view uint64, id uint32) *messages.Prepare {
 		return &messages.Prepare{
 			Msg: &messages.Prepare_M{
 				View:      view,
 				ReplicaId: id,
 				Request:   request,
 			},
-			ReplicaUi: prepareUIBytes,
 		}
 	}
 
 	var replyChan chan *messages.Reply
 
 	prepare := makePrepareMsg(view, id)
 
-	mock.On("uiAcceptor", prepare).Return(false, fmt.Errorf("Invalid UI")).Once()
+	mock.On("uiVerifier", prepare).Return((*usig.UI)(nil), fmt.Errorf("UI not valid")).Once()
 	_, err := handle(prepare)
 	assert.Error(t, err, "UI check failed")
 
-	mock.On("uiAcceptor", prepare).Return(false, nil).Once()
+	mock.On("uiVerifier", prepare).Return(ui, nil).Once()
+	mock.On("uiAcceptor", id, ui).Return(false, nil).Once()
 	mock.On("requestHandler", request, true).Return(replyChan, false, nil).Once()
 	new, err := handle(prepare)
 	assert.NoError(t, err)
 	assert.False(t, new, "UI already processed")
 
-	prepare = makePrepareMsg(view+1, primaryID(n, view))
-	mock.On("uiAcceptor", prepare).Return(true, nil).Once()
-	mock.On("uiCommitter", prepare).Once()
+	otherView := view + 1
+	otherPrimary := primaryID(n, view+1)
```

Comment:
(... nitpicking) using `otherView` instead of `view+1`?

Reply:
:+1: Thanks!